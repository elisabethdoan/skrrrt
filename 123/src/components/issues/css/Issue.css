#issue {
    border-style: groove;
    margin-bottom: 5px;
    width: 60%;
    background-color: #E2432A;
    border-radius: 1%;
    overflow-wrap: break-word;
    hyphens: auto;
  }
  
  #issueDescription{
    text-align: left;
    margin-top: 0;
    margin-left: 1vh;
  }
  
  #title{
    margin-bottom: 2vh;
    margin-top: 2vh;
  }

  #issue > p {
    text-align: left;
    margin-left: 1vh;
    margin-top: 0;
    margin-bottom: 1vh;
  }

  #line{
    border: 1px dashed;
  }

  @media only screen and (max-width: 600px) {
    #issue {
      width: 90%;
    }
  }