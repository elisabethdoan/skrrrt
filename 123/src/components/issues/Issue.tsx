import React from 'react';
import './css/Issue.css';

interface Props {
  issue: {
    id: number;
    title: string;
    body: string;
    state: 'open' | 'closed';
    labels: {
      name: string;
    }[];
    // eslint-disable-next-line
    created_at: string; // ISO 8601 date-time string
    // eslint-disable-next-line
    closed_at: string; // ISO 8601 date-time string
  };
}
/**
 * This component is a textual representation of a single issue.
 */
const Issue: React.FC<Props> = ({ issue }) => {
  console.log(issue.state);
  const creationDate = new Date(issue.created_at).toLocaleString();
  let closingDate;
  if (issue.closed_at !== null) {
    closingDate = new Date(issue.closed_at).toLocaleString();
  } else {
    closingDate = null;
  }
  const labelString = issue.labels.map((label) => label.name).toString();
  return (
    <div id="issue">
      <h3 id="title">{`${issue.title} #${issue.id}`}</h3>
      <hr id="line" />
      <br />
      <p>
        <b>Description: </b>
        {issue.body}
      </p>
      <p>
        <b>State: </b>
        {issue.state}
      </p>
      <p>
        <b>Labels: </b>
        {labelString}
      </p>
      <p>
        <b>Created at: </b>
        {creationDate}
      </p>
      <p>
        {issue.closed_at ? <b>Closed at: </b> : null}
        {closingDate}
      </p>
    </div>
  );
};

export default Issue;
