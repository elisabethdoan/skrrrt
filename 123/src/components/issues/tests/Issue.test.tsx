import React from 'react';
import renderer from 'react-test-renderer';
import Issue from '../Issue';

const testIssue = {
  title: 'TestIssue',
  iid: 1,
  description: 'Dette er et testIssue',
  state: 'closed',
  labels: [],
  // eslint-disable-next-line
  created_at: '2021-09-28T13:00:36.099+02:00',
  // eslint-disable-next-line
  closed_at: "2021-09-29T13:00:36.099+02:00"
};

test('Tests that rendering IssueChart matches snapshot,', () => {
  const tree = renderer.create(<Issue issue={testIssue} />).toJSON();
  expect(tree).toMatchSnapshot();
});
