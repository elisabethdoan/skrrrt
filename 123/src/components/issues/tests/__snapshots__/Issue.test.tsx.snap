// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests that rendering IssueChart matches snapshot, 1`] = `
<div
  id="issue"
>
  <h3
    id="title"
  >
    TestIssue #1
  </h3>
  <hr
    id="line"
  />
  <br />
  <p>
    <b>
      Description: 
    </b>
    Dette er et testIssue
  </p>
  <p>
    <b>
      State: 
    </b>
    closed
  </p>
  <p>
    <b>
      Labels: 
    </b>
    
  </p>
  <p>
    <b>
      Created at: 
    </b>
    28.9.2021, 13:00:36
  </p>
  <p>
    <b>
      Closed at: 
    </b>
    29.9.2021, 13:00:36
  </p>
</div>
`;
